src/V3Ast.h

```diff
@@ -2108,71 +2109,115 @@ class AstNode VL_NOT_FINAL {
     // handle a single (or a few) node types, as it's easier to write, but more importantly, the
     // dispatch to the operation function in 'foreach' should be completely predictable by branch
     // target caches in modern CPUs, while it is basically unpredictable for VNVisitor.
-    template <typename T_Node>
-    void foreach (std::function<void(T_Node*)> f) {
-        static_assert(checkTypeParameter<T_Node>(), "Invalid type parameter 'T_Node'");
+    template <typename Func>
```

Comment:
Lest's call the template argument `Callable` as in some cases it is not a function

Reply:
Not quite sure whether you ignored this intentionally, or if I misunderstood something

Reply:
That's my mistake, I will fix soon.

Reply:
Thank you